assets/wizards/subscriptions/index.js

```diff
@@ -129,33 +155,116 @@ class SubscriptionsWizard extends Component {
 		);
 	}
 
+	onSubscriptionChange = subscription => {
+		this.setState( prevState => ( {
+			subscriptions: {
+				...prevState.subscriptions,
+				[ subscription.id ]: subscription,
+			},
+		} ) );
+	};
+
 	/**
 	 * Render.
 	 */
 	render() {
-		const { subscriptions, editing, choosePrice } = this.state;
-
-		if ( !! editing ) {
-			return (
-				<EditSubscriptionScreen
-					subscription={ editing }
-					onChange={ subscription => this.setState( { editing: subscription } ) }
-					onClickSave={ subscription => this.saveSubscription( subscription ) }
-					onClickCancel={ () => this.setState( { editing: false } ) }
-				/>
-			);
-		} else {
-			return (
-				<ManageSubscriptionsScreen
-					subscriptions={ subscriptions }
-					choosePrice={ choosePrice }
-					onClickEditSubscription={ subscription => this.setState( { editing: subscription } ) }
-					onClickDeleteSubscription={ subscription => this.deleteSubscription( subscription.id ) }
-					onClickChoosePrice={ () => this.toggleChoosePrice() }
-				/>
-			);
-		}
+		const { pluginRequirements } = this.props;
+		const { subscriptions, choosePrice } = this.state;
+		return (
+			<HashRouter hashType="slash">
+				<Switch>
+					{ pluginRequirements }
+					<Route
+						path="/"
+						exact
+						render={ routeProps => (
+							<ManageSubscriptionsScreen
+								headerText={
+									Object.values( subscriptions ).length
+										? __( 'Any more subscriptions to add?' )
+										: __( 'Add your first subscription' )
+								}
+								subHeaderText={ __(
+									'Subscriptions can provide a stable, recurring source of revenue'
+								) }
+								subscriptions={ Object.values( subscriptions ) }
+								choosePrice={ choosePrice }
+								onClickDeleteSubscription={ subscription =>
+									this.deleteSubscription( subscription.id )
+								}
+								onClickChoosePrice={ () => this.toggleChoosePrice() }
+								buttonText={
+									subscriptions.length
+										? __( 'Add another subscription' )
+										: __( 'Add a subscription' )
+								}
+								buttonAction="#/create"
+							/>
+						) }
+					/>
+					<Route
+						path="/edit/:id"
+						render={ routeProps => {
+							return (
+								<EditSubscriptionScreen
+									headerText={ __( 'Edit subscription' ) }
+									subHeaderText={ __( 'You are editing an existing subscription' ) }
+									subscription={ subscriptions[ routeProps.match.params.id ] || {} }
+									onChange={ this.onSubscriptionChange }
+									onClickSave={ subscription =>
+										this.saveSubscription( subscription ).then( () => this.refreshSubscriptions() )
```

Comment:
After saving a subscription, we should be dropped off on the Subscriptions Management screen (`#/`). I was dropped off on the Edit Subscription screen (`#/edit/:id`) after saving. Not a really big deal, but right now it forces a weird flow where you have to Save and then Cancel after saving to go to the Manage Subscriptions screen.